File: Q2-Assignments/Base on Java/OthobaAutomationJava/src/main/java/pages/ProductPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;

public class ProductPage {
    WebDriver driver;

    public ProductPage(WebDriver driver) {
        this.driver = driver;
    }

    // Locators
    By searchBox = By.id("search-box");
    By searchButton = By.id("search-submit");
    By productLink = By.xpath("//div[contains(@class, 'product-name')]");

    // Actions on Product Page
    public void searchProduct(String productName) {
        driver.findElement(searchBox).sendKeys(productName);
        driver.findElement(searchButton).click();
    }

    public void selectProduct() {
        driver.findElement(productLink).click();
    }
}

File: Q2-Assignments/Base on Python/pages/product_page.py
from selenium.webdriver.common.by import By

class ProductPage:
    def __init__(self, driver):
        self.driver = driver
        self.add_to_cart_button = (By.ID, 'addToCart')

    def click_add_to_cart(self):
        self.driver.find_element(*self.add_to_cart_button).click()
